CruCentralCoast/Events/EventsVC.swift

```diff
@@ -48,47 +57,102 @@ class EventsTVC: UITableViewController {
-                let cell = tableView.cellForRow(at: indexPath)
-                UIView.animate(withDuration: 0.2, animations: {
-                    cell?.transform = CGAffineTransform(scaleX: 0.9, y: 0.9)
-                })
+                self.longPressIndexPath = indexPath
+                self.longPressGestureBegan(indexPath: indexPath)
             }
         } else if gestureRecognizer.state == .ended {
             if let indexPath = tableView.indexPathForRow(at: touchPoint) {
-                let cell = tableView.cellForRow(at: indexPath)
-                guard let vc = UIStoryboard(name: "Events", bundle: nil).instantiateViewController(withIdentifier: "EventDetailsVC") as? EventDetailsVC else {
-                    assertionFailure("Probably used the wrong storyboard name or identifier here")
-                    return
+                if indexPath == self.longPressIndexPath {
+                    self.longPressGestureEnded(indexPath: indexPath)
+                } else {
+                    self.longPressGestureCancelled(indexPath: self.longPressIndexPath)
                 }
-                UIView.animate(withDuration: 0.05, animations: {
-                    cell?.transform = CGAffineTransform.identity
-                }, completion: { (finish) in
-                    vc.configure(with: self.dataArray[indexPath.item])
-                    vc.transitioningDelegate = self
-                    self.selectedCell = self.tableView.cellForRow(at: indexPath) as? EventsTableCell
-                    self.navigationController?.present(vc, animated: true, completion: nil)
-                })
             }
         } else if gestureRecognizer.state == .cancelled {
+            self.longPressCancelled = true
             if let indexPath = tableView.indexPathForRow(at: touchPoint) {
-                let cell = tableView.cellForRow(at: indexPath)
-                UIView.animate(withDuration: 0.2, animations: {
-                    cell?.transform = CGAffineTransform.identity
-                })
+                self.longPressGestureCancelled(indexPath: indexPath)
+            }
+        }
+        else if gestureRecognizer.state == .failed {
+            self.longPressFailed = true
+            if let indexPath = tableView.indexPathForRow(at: touchPoint) {
+                self.longPressGestureCancelled(indexPath: indexPath)
             }
         }
     }
+    
+    func longPressGestureBegan(indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath)
+        UIView.animate(withDuration: 0.2, animations: {
+            cell?.transform = CGAffineTransform(scaleX: 0.95, y: 0.95)
+        })
+    }
+    
+    func longPressGestureEnded(indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath)
+        guard let vc = UIStoryboard(name: "Events", bundle: nil).instantiateViewController(withIdentifier: "EventDetailsVC") as? EventDetailsVC else {
```

Comment:
Might want to use the instantiateViewController(_ classType: T.Type) method that we created to make this cleaner

Reply:
Im having trouble with this part. The method you are talking about is a UIStoryBoard extension. When I call it like this:

guard let vc = UIStoryBoard(name: "Event", bundle: nil).instantiateViewController(EventDetails.self) else { return }

I did that and getting this error: 
Initializer for conditional binding must have Optional type, not 'EventDetailsVC'

must be calling it the wrong way.